### PR TITLE
feat(Form.Handler): enhance async behaviour

### PR DESCRIPTION
**Pull request by tujoworker**
- fix: ensure timeout executes always
- Swap out docs for `Form.Element` with `Form.Handler`
- Ensure the user can set the form in pending mode while an async validation is on going
- Ensure only the relevant value will be evaluated
- Ensure context value is always in sync
- Make field submit status (pending/error) more stable
- Ensure Provider onSubmit gets not called several times when sync (not async)
- Rename `asyncBehaviorTimeout` to `asyncSubmitTimeout` for better descriptive name

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/info.mdx [hunk 1]
```diff
@@ -7,10 +7,7 @@ import AsyncChangeExample from './parts/async-change-example.mdx'
 
 ## Description
 
-The `Form.Handler` component provides both the [Form.Element](/uilib/extensions/forms/extended-features/Form/Element) and a HTML form element.
-
-- It ensures, users can press **enter key** while focusing on an input field.
-- It calls `preventDefault` by default.
+The `Form.Handler` component provides both the [Form.Element](/uilib/extensions/forms/extended-features/Form/Element/info/) and a HTML form element.
```
Comment (joakbjerk) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/info.mdx:10]: This link is dead now 🪦

**Review by joakbjerk — CHANGES_REQUESTED** [review]

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/info.mdx (hunk 1 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/info.mdx:10]: Almost yeah. The heading is missing. We can create a one-pager instead. I just don't think we need to list `Form.Element` as a thing devs need to use/know off.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/demos.mdx [hunk 2]
```diff
@@ -62,3 +58,23 @@ when the browser reloads. The data is stored until the session storage is invali
...
+
+This example demonstrates how to use async validation with an async `onSubmit` and async `onChange` event for both the `Form.Handler` and a field itself.
+
+- While you write, an async validation request is simulated to check if the input is valid. If it's not, an error message will be shown.
+
+- During validation, only the relevant value will be evaluated. This means, when the delayed validation is done, and the value has changed, the validation result will be omitted.
+
+- You can press enter to submit the form while you write. But only a string of `valid` will be accepted to emit the form `onSubmit` and `onChange`.
+
+- You can starting wiring, wait a second or two and remove the whole text again and blur the field. The async validation return will be omitted and the "required" error message will be shown.
```
Comment (joakbjerk) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/extended-features/Form/Handler/demos.mdx:72]:
```suggestion
- You can start writing, wait a second or two and remove the whole text again and blur the field. The async validation return will be omitted and the "required" error message will be shown.
```

**Review by joakbjerk — APPROVED** [review]
Looking good 👑

**Pull request merged**